src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ParameterBinder.cs

```diff
@@ -185,14 +185,14 @@ public class ParameterBinder
             await modelBinder.BindModelAsync(modelBindingContext);
 
             var modelBindingResult = modelBindingContext.Result;
-            if (modelBindingResult.IsModelSet)
-            {
-                _validator.Validate(
-                    actionContext,
-                    modelBindingContext.ValidationState,
-                    modelBindingContext.ModelName,
-                    modelBindingResult.Model);
-            }
+
+            _parameterValidator.Validate(
+                actionContext,
+                parameter,
+                metadata,
+                modelBindingResult.IsModelSet,
+                modelBindingResult.Model,
+                modelBindingContext);
```

Comment:
Ah ok we already have this class. I feel like most of the new logic that's in your `IParameterBinder` should be here.

Reply:
OK done.